src/fairness/writer/data_writer_stdout.hpp

```diff
@@ -0,0 +1,59 @@
+/************************************************************\
+ * Copyright 2021 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#include <sqlite3.h>
+#include <cerrno>
+
+#include "src/fairness/weighted_tree/weighted_walk.hpp"
+#include "src/fairness/writer/data_writer_base.hpp"
+
+using namespace Flux::accounting;
+
+namespace Flux {
+namespace writer {
+
+/*!  Base data writer class.
+ */
+class data_writer_stdout_t : public data_writer_base_t {
```

Comment:
@cmoussa1: what is the abstract interface `data_writer_stdout_t` implement? Is that `int write_acct_info` in the base class? If so, that method in `data_writer_base_t` should be marked as `virtual` with `=0`? 

As is, it is not clear why `data_writer_stdout_t` is a kind of `data_writer_base_t`. A typical pattern I have seen is a basic set of `writer` operations are abstracted out in the base class and each derived writer subclasses implement those.

Reply:
Yeah, I understand - thanks for this clarification. I suppose I should have appended a `TODO` to the description of the PR, as I still had to implement the `write_acct_info ()` method from the sub class - I still had that work left to do, I'm sorry for not making that more clear.

My plan was to overload the `write_acct_info ()` method in `data_writer_stdout_t` with the parameters specific to this subclass, like the `delimiter` argument. Then I could make the helper functions in `data_writer_stdout_t` private methods and have `flux_shares.cpp` just call `write_acct_info ()`. Does this implementation make sense and follow that pattern?